### PR TITLE
Fetch data project page

### PR DESCRIPTION
**Pull request by Dank-del**

### CODE REVIEW

**Comment by Dank-del** [comment]
Screenshot 2023-12-29 at 4 16 34 PM
Interesting how it does build on local machine.. does it fail due to some env var here?

File: apps/web/src/app/project/[slug]/components/projectDetailsPageHeader.tsx [hunk 1]
```diff
@@ -11,22 +10,43 @@ import {
...
 const ProjectDetailsPageHeader = ({
   project,
 }: {
-  project: ProjectPropsType;
+  project: ProjectPropsType | any;
 }) => {
+  console.log(project);
+  const { event, setEvent } = useProjectEventStore();
+
+  React.useEffect(() => {
```
Comment (dhruvsol) [inline apps/web/src/app/project/[slug]/components/projectDetailsPageHeader.tsx:41]: why are you seting this here ?

Comment (dhruvsol) [reply apps/web/src/app/project/[slug]/components/projectDetailsPageHeader.tsx:41]: also dont set any state directly in useEffect cause a lot of re-render even with specific things pass in the array

File: apps/web/src/app/project/[slug]/components/tabs/projectContributorsTab.tsx [hunk 2]
```diff
@@ -1,17 +1,197 @@
...
+                      </TableCell>
+                      <TableCell>
+                        <Text color='primary' className='h-8'>
+                          {formatDistanceToNow(contribution.createdAt)} ago
+                        </Text>
+                      </TableCell>
+                    </TableRow>
+                  )
+                })}
+                <div
```
Comment (dhruvsol) [inline apps/web/src/app/project/[slug]/components/tabs/projectContributorsTab.tsx:141]: use <Spinner/>

File: apps/web/src/app/project/[slug]/components/tabs/projectDetailsTab.tsx [hunk 3]
```diff
@@ -277,51 +280,68 @@ const InteractionSidebar = ({
...
-            // @ts-ignore
-            slides={fetchedProjectDetails[0]?.slides}
-          />
-          <InteractionSidebar
-            socials={{
-              // @ts-ignore
-              github: fetchedProjectDetails[0].githubLink,
+          <div className='flex flex-col'>
+            <ImagesCarousel
               // @ts-ignore
```
Comment (dhruvsol) [inline apps/web/src/app/project/[slug]/components/tabs/projectDetailsTab.tsx:305]: whats the type issue your getting ?

File: apps/web/src/app/project/[slug]/actions.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+ * @param userId - The ID of the user creating the comment.
+ * @param projectId - The ID of the project the comment belongs to.
+ * @returns The created comment.
+ * @throws Error if the user is not logged in.
+ */
+const makeComment = async (comment: string, userId: string, projectId: string) => {
+    if (!userId) {
+        throw new Error('User not logged in');
+    }
+    if (!comment) {
```
Comment (dhruvsol) [inline apps/web/src/app/project/[slug]/actions.ts:15]: throw a error in this case so we can handle it on client side pop a user by toast or something

**Review by dhruvsol — CHANGES_REQUESTED** [review]

File: apps/web/src/app/project/[slug]/components/projectDetailsPageHeader.tsx (hunk 1 above)
Comment (Dank-del) [reply apps/web/src/app/project/[slug]/components/projectDetailsPageHeader.tsx:41]: this is for default event state

File: apps/web/src/app/project/[slug]/components/tabs/projectDetailsTab.tsx (hunk 3 above)
Comment (Dank-del) [reply apps/web/src/app/project/[slug]/components/tabs/projectDetailsTab.tsx:303]: Screenshot 2024-01-03 at 2 03 47 PM
Screenshot 2024-01-03 at 2 04 05 PM

Comment (Dank-del) [reply apps/web/src/app/project/[slug]/components/tabs/projectDetailsTab.tsx:303]: Screenshot 2024-01-03 at 2 06 12 PM
But I do have slides in it, its just a slightly modified fetch action which already was there before i worked on it.

File: packages/ui/components/ui/MenuList/MenuList.tsx [hunk 5]
```diff
@@ -127,7 +127,7 @@ const MenuItem = ({
           {leftIcon && (
             <Icon name={leftIcon} stroke="inherit" height={20} width={20} />
           )}
-          <Text className="b1 md:l2-light" color={'inherit'}>
+          <Text className="b1 md:l2-light-light" color={'inherit'}>
```
Comment (dhruvsol) [inline packages/ui/components/ui/MenuList/MenuList.tsx:130]: there is no l2-light-light

Comment (dhruvsol) [reply packages/ui/components/ui/MenuList/MenuList.tsx:130]: change to l2-light

**Review by dhruvsol — CHANGES_REQUESTED** [review]

**Pull request merged**